storm-core/src/jvm/backtype/storm/utils/ShellProcess.java

```diff
@@ -186,4 +163,35 @@ public String getProcessInfoString() {
     public String getProcessTerminationInfoString() {
         return String.format(" exitCode:%s, errorString:%s ", getExitCode(), getErrorsString());
     }
+
+    private Optional<String> getErrorStreamMessage() {
+        int offset = 0;
+        if (processErrorStream != null) {
+            ByteArrayOutputStream out = null;
+            try {
+                do {
+                    int bufferSize = processErrorStream.available();
+                    if (bufferSize > 0) {
+                        if (out == null) {
+                            out = new ByteArrayOutputStream();
+                        }
+                        byte[] errorReadingBuffer = new byte[bufferSize];
+                        final int read = processErrorStream.read(errorReadingBuffer, 0, bufferSize);
+                        out.write(errorReadingBuffer, offset, bufferSize);
+                        offset += read;
+                    }
+                } while (processErrorStream.available() > 0);
+
+                if (offset > 0)
+                    return of(new String(out.toByteArray()));
+            } catch (Exception e) {
+                return of("(Unable to capture error stream)");
+            } finally {
+                if (out != null) {
+                    IOUtils.closeQuietly(out);
+                }
+            }
+        }
+        return Optional.absent();
```

Comment:
@reembs This project ussually do not use Optional<>... so I would just return an empty string